occlusion.html

```diff
@@ -0,0 +1,708 @@
+        mat4.lookAt(viewMatrix, eyePosition, vec3.fromValues(0, 0, 0), vec3.fromValues(0, 1, 0));
+
+        var viewProjMatrix = mat4.create();
+        mat4.multiply(viewProjMatrix, projMatrix, viewMatrix);
+
+        // top down assist camera
+        var projMatrixAssistCam = mat4.create();
+        mat4.perspective(projMatrixAssistCam, Math.PI / 2, gl.drawingBufferWidth / gl.drawingBufferHeight, 0.1, 10.0);
+
+        var viewMatrixAssistCam = mat4.create();
+        var eyePositionAssistCam = vec3.fromValues(3, 5, -2);
+        mat4.lookAt(viewMatrixAssistCam, eyePositionAssistCam, vec3.fromValues(3, 0, -2), vec3.fromValues(0, 0, -1));
+
+        var viewProjMatrixAssistCam = mat4.create();
+        mat4.multiply(viewProjMatrixAssistCam, projMatrixAssistCam, viewMatrixAssistCam);
+
+        gl.useProgram(programAssistCam);
+        gl.uniformMatrix4fv(viewProjMatrixLocationAssistCam, false, viewProjMatrixAssistCam);
+        gl.useProgram(null);
+
+        var lightPosition = vec3.fromValues(1, 1, 2);
+
+        var modelMatrix = mat4.create();
+        var rotateXMatrix = mat4.create();
+        var rotateYMatrix = mat4.create();
+
+        var sceneUniformData = new Float32Array(24);
+        sceneUniformData.set(viewProjMatrix);
+        sceneUniformData.set(eyePosition, 16);
+        sceneUniformData.set(lightPosition, 20);
+
+        var sceneUniformBuffer = gl.createBuffer();
+        gl.bindBufferBase(gl.UNIFORM_BUFFER, 0, sceneUniformBuffer);
+        gl.bufferData(gl.UNIFORM_BUFFER, sceneUniformData, gl.STATIC_DRAW);
+
+        var assistCamUniformData = new Float32Array(24);
+        assistCamUniformData.set(viewProjMatrixAssistCam);
+        assistCamUniformData.set(eyePositionAssistCam, 16);
+        assistCamUniformData.set(lightPosition, 20);
+
+        var assistCamUniformBuffer = gl.createBuffer();
+        gl.bindBufferBase(gl.UNIFORM_BUFFER, 1, assistCamUniformBuffer);
+        gl.bufferData(gl.UNIFORM_BUFFER, assistCamUniformData, gl.STATIC_DRAW);
+
+        var image = new Image();
+        var firstFrame = true;
+        var isDrawNodes = new Array(nodes.length);
+        for (var i = 0, len = isDrawNodes.length; i < len; ++i) {
+            isDrawNodes[i] = true;
+        }
+        var invisibleNodesCount = 0;
+
+        image.onload = function() {
+            var texture = gl.createTexture();
+            gl.activeTexture(gl.TEXTURE0);
+            gl.bindTexture(gl.TEXTURE_2D, texture);
+
+            gl.pixelStorei(gl.UNPACK_FLIP_Y_WEBGL, true);
+            gl.texParameteri(gl.TEXTURE_2D, gl.TEXTURE_MAG_FILTER, gl.LINEAR);
+            gl.texParameteri(gl.TEXTURE_2D, gl.TEXTURE_MIN_FILTER, gl.LINEAR_MIPMAP_LINEAR);
+            gl.texParameteri(gl.TEXTURE_2D, gl.TEXTURE_WRAP_S, gl.REPEAT);
+            gl.texParameteri(gl.TEXTURE_2D, gl.TEXTURE_WRAP_T, gl.REPEAT);
+
+            var levels = levels = Math.floor(Math.log2(Math.max(this.width, this.height))) + 1;
```

Comment:
Ha! I was thinking "why did he do levels = levels" and then I realized that's from my code :)